### PR TITLE
Remove `source-map` dependency from terser bundle

### PR DESCRIPTION
**Pull request by sbc100**
When updating terser to 4.1.0 in #17035 I inadvertently introduced a
dependency on the node source-map module.  Rebuilding after patching
our terser fork removed this:

https://github.com/emscripten-core/terser/commit/b3ace3b63703d23316b224ef00dd32669369a948

This happens to be available during testing because it one of our dev
decpendencies but we not include it in our end-user/production
dependencies.

This kind of error is hard to find since the `node_modules` for our test
setup is the same as the node_modules for emscripten itself.  It migth
be worth attempting to tease them appart if this kind of thing ever
shows up again.

### CODE REVIEW

File: third_party/terser/README.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ To make changes to this code please submit patches to
 https://github.com/emscripten-core/terser/ and then re-create this bundle
 using the following steps:
 
-  $ CI=1 npm run build
+  $ npx rollup -c
```
Comment (kripken) [inline third_party/terser/README.md:16]: Intentional change? What is the difference?

**Review by kripken — APPROVED** [review]

File: third_party/terser/README.md (hunk 1 above)
Comment (sbc100) [reply third_party/terser/README.md:16]: (and I can then remove the comment below)

**Comment by kripken** [comment]
Looks like this affected binaryen CI, https://github.com/WebAssembly/binaryen/runs/6613053782?check_suite_focus=true which I guess will be fixed by this.

**Comment by sbc100** [comment]
Yes, this is an odd case where the only effect users are those that are not setup to run emscripten tests... so by definition it doesn't get caught be our testing process :(

**Comment by kripken** [comment]
Yeah... I guess it makes sense that emsdk CI would find it as that really tests the entire user flow. I agree it might not be worth doing anything more there, as such breakage will be 100% caught before a release.

**Comment by sbc100** [comment]
Its also notable that users who happen to have `source-map` install anywhere in their node path would also not have seen this issue..  the emcc JS tools are not insulated from the wider node module environment.

**Pull request merged**